### PR TITLE
Stats Revamp: Fix Period Store State when navigating between 1st and 2nd levels

### LINKED ISSUE #18983: [iOS] Switching between Stats tabs causes incorrect stats data to be shown
### Expected behavior

If I'm on the Insights tab, then switch to another tab (e.g. Days, Weeks, Months, Years), and then return to Insights, the app should still show the correct stats (e.g. for Views & Visitors).

### Actual behavior

When I switch to another tab and return to Insights, the data shown on Insights is now incorrect. (It looks like that app shows in Insights the data from the previously selected tab.

### Steps to reproduce the behavior

1. Open the My Site tab
2. Do either of the following
  - On the Home tab, tap the Stats button
  - On the Menu tab, tap the Stats list item
4. On the Insights tab, take note of the values shown on the screen (e.g. Views or Visitor values)
5. Switch to another tab (e.g. Weeks or Months or Years)
6. Switch back to the Insights tab 
7. Notice that the values are not the same as those noted in Step 2

The workaround is to exit Stats, return to the My Site screen, and navigate to Stats again.

### Additional Information

This is a new regression in 20.2, I tested and it's not present in 20.1

### PR DESCRIPTION
**Pull request by sla8c**
This PR adds functionality so that when a user navigates to the main Insights Screen the PeriodStore will be set to .day granularity and a fresh call will be made for its data state.

Fixes #18983

This is quickest win solution for now.  What is occurring is:
 * PeriodStore is a singleton
 * When tapping any of the Period stats such as Day / Week / Month / Year, the singleton PeriodStore internally holds the state of the interval selected such as Month or Year
 * When navigating back to the main insights screen, the ViewsVisitors chart reads from PeriodStore again however PeriodStore data is set for 'Year' or 'Month' ie whatever was selected

I tried another approach where I created a 2nd instance of PeriodStore to be used exclusively by the top level insight screen but this didn't work because the Flux events for PeriodStore are still the same so both PeriodStore still receive the same Flux events and are left in the same state.

In the future depending on what decisions are taken with Period Stats (Days / Weeks / Months) screens this could be further optimised and an enhanced Stats DataStore that accommodates mixed period scenarios could be worked on.

To test:
1. enable both of the new stats feature flags statsNewAppearance and statsNewInsights via `AppConfiguration.showsStatsRevampV2`
2. From top level Stats Insights screen take note of the Views / Visitors count
3. Tap on one of the Period stats tabs such as Weeks, Months, Years.
4. Navigate back to the insights tab.  The Views / Visitors count should be the same as in step 2
5. Switch sites and continue to test this for other sites

## Regression Notes
1. Potential unintended areas of impact2. 
Stats Insights

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested

4. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift [hunk 1]
```diff
@@ -75,6 +75,8 @@ class SiteStatsInsightsViewModel: Observable {
     }
 
     func startFetchingPeriodOverview() {
+        self.lastRequestedPeriod = StatsPeriodUnit.day
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift]: This change may not be needed.

`self.lastRequestedPeriod = StatsPeriodUnit.day` is set in `init()` and never changed. I tested with and without this change and couldn't find it affecting the outcome.

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift [hunk 2]
```diff
@@ -107,6 +107,8 @@ class SiteStatsInsightsDetailsViewModel: Observable {
                 }
                 insightsReceipt = insightsStore.query(storeQuery)
 
+                let date = selectedDate ?? StatsDataHelper.currentDateForSite()
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift]: This change may not be needed. 

With or without this change "Comments Total" details appear in the same way, no matter if I switch between Days/Weeks/Months/Years.

**Review by staskus — COMMENTED** [review]
Nice that this fix is already made, @sla8c! 👍

I'm sure that it's hard to remember any details after couple of months so let me know how you want to better approach this pull request.

1. I merged this feature branch with the latest trunk
2. I tested every stats card and checked numbers while switching between Insights <-> Days/Weeks/Months/Years tabs
3. I tried to understand what each line of the PR changes. It looks like calling `startFetchingPeriodOverview` does the trick. I couldn't find how other changes influence the issue. I'm not sure if you remember any details about that? 

Thank you! 🚀

**Comment by guarani** [comment]
I've removed myself as a reviewer now that @staskus is reviewing this. Please let me know though if I can help out with this or other PR reviews.

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift]: Thanks for looking at this with a fresh set of eyes! I agree as well that I don't think this is needed.  I've removed

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift (hunk 2 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift]: Ditto - thanks for looking at this with a fresh set of eyes! I agree as well that I don't think this is needed.  I've removed

**Review by staskus — APPROVED** [review]
Thanks for the fixes 👍 I'm glad that the suggestions were valuable.

**Pull request merged**